core/uemis-downloader.c

```diff
@@ -603,7 +603,7 @@ static bool uemis_get_answer(const char *path, char *request, int n_param_in,
 			free(ans_path);
 			return false;
 		}
-		if (read(ans_file, tmp, 100) < 0) {
+		if (read(ans_file, tmp, 100) < 100) {
```

Comment:
I haven't played with the UEMIS downloader in ages because all of the Uemis dive computers that I had access to have long since died.
Every time I look at the code my eyes bleed...
I am not sure if we are guaranteed that a 100 characters can be read here. Reading the code we only ever appear to look at the first 4 bytes of this buffer? Am I missing something?
So this one I don't think is necessarily right and it could change behavior of Subsurface if indeed we get fewer than 100 bytes back in a valid read...

Reply:
You are right, checking against 100 is definitely wrong. I was too quick and only saw the tmp[100] in the debug code before not even realising that that is writing rather than reading. I also have to admit I confused the return value of read with fread. But what would be the correct value, maybe 3 (as we are using up to tmp[2] below?

Reply:
I re-read the code. I believe we need to get at least 3 bytes to be able to process the buffer.